src/DIRAC/DataManagementSystem/Client/FTS3Job.py

```diff
@@ -22,21 +24,61 @@
 # default pycurl. See https://its.cern.ch/jira/browse/FTS-261
 from fts3.rest.client.request import Request as ftsSSLRequest
 
-from DIRAC.Resources.Storage.StorageElement import StorageElement
-
-from DIRAC.FrameworkSystem.Client.Logger import gLogger
-from DIRAC.FrameworkSystem.Client.TokenManagerClient import gTokenManager
-
-from DIRAC.Core.Utilities.ReturnValues import S_OK, S_ERROR
 from DIRAC.Core.Utilities.DErrno import cmpError
-
 from DIRAC.Core.Utilities.JEncode import JSerializable
+from DIRAC.Core.Utilities.ReturnValues import S_ERROR, S_OK
 from DIRAC.DataManagementSystem.Client.FTS3File import FTS3File
+from DIRAC.FrameworkSystem.Client.Logger import gLogger
+from DIRAC.FrameworkSystem.Client.TokenManagerClient import gTokenManager
+from DIRAC.Resources.Storage.StorageElement import StorageElement
 
 # 3 days in seconds
 BRING_ONLINE_TIMEOUT = 259200
 
 
+# Create a TTL cache: max 1 item, expires after 3600 seconds (1 hour)
+scitag_cache = TTLCache(maxsize=10, ttl=3600)
+
+
+@cached(scitag_cache)
```

Comment:
Sorry I forgot before but given that this agent is multithread, you probably want to add a cache lock.

Reply:
Done.